### ISSUE TITLE
Terminal.app title

### ISSUE THREAD

**Issue by jscheid** [casey/just#1923]
I'm not sure if this is an issue only with my terminal (Terminal.app 2.14 on macOS Sonoma), my shell (zsh 5.9), my termcap or the following recipe:

```justfile
psql *REST:
    #!/usr/bin/env sh
    set -eou pipefail
    #echo -en "\033]0;\a"
    export PGAPPNAME="psql[$USER@$(hostname)]"
    psql \
      -h localhost \
      -p 55432 \
      -U myuser \
      -d mydb \
      "$@"
```

... but when running this recipe (with `j psql`, just 1.24.0) the Terminal tab title only says "j" as seen in this screenshot:
<img width="269" alt="Screenshot 2024-02-22 at 00 59 42" src="https://github.com/casey/just/assets/3514/1d0a3abc-1ae7-4cbf-a015-81d8d652b3fd">

I often run different recipes in different tabs and so this isn't quite ideal because I can't tell them apart by the title when they all say "j".

By un-commenting the `echo` line in the above recipe the title becomes much more useful:
![image](https://github.com/casey/just/assets/3514/7e94206d-ab68-415c-8196-882c16303332)

I could put this into all my recipes but that's a bit messy and easy to forget. I'm wondering if there's anything that could be done in `just` to make this work better?

**Comment by casey** [comment]
Interesting! What is `\033]0;\a`? Is it a terminal escape code to change the terminal title? I can't find documentation for it, and I can't get it to work on my mac. If there was a non-hacky way to get it working cross-platform, I definitely see how it could be useful.

**Comment by jscheid** [comment]
> What is \033]0;\a? Is it a terminal escape code to change the terminal title? I can't find documentation for it

Yes, here is some documentation: https://tldp.org/HOWTO/Xterm-Title-3.html

I've found a repro that doesn't require you to have a PostgreSQL server running:

```justfile
test:
    docker run linuxserver/emulatorjs
```

Run `just test` in Terminal.app, wait for the container to boot, then wait another 10 seconds and the tab title should show just `just`. Add the magic incantation from above and it will show a more useful title (and keep showing it after 10s.)

This being said, I discovered that `bash -c 'docker run linuxserver/emulatorjs'` has exactly the same problem, it too will just show `bash` after some time. I thought it was something that `just` was doing but it appears to be a (mis)feature in Terminal.app instead.

> If there was a non-hacky way to get it working cross-platform, I definitely see how it could be useful.

I'm currently trying to find a workaround with a custom "fake" shell that runs the echo before invoking the command itself but not having much luck. I'll keep trying but maybe you have an idea?

Here's my current attempt (it's not working, it seems the escape sequence is ignored when it runs from "inside" the shell):

```sh
#!/bin/sh

echo -en "\033]0;\a"
exec sh "$@"
```
